sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -327,6 +327,262 @@ public CosmosClientBuilder contentResponseOnWriteEnabled(boolean contentResponse
         return this;
     }
 
+    /**
+     * Sets the default GATEWAY connection configuration to be used.
+     *
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder gatewayMode() {
+        this.gatewayConnectionConfig = GatewayConnectionConfig.getDefaultConfig();
+        return this;
+    }
+
+    /**
+     * Sets the GATEWAY connection configuration to be used.
+     *
+     * @param gatewayConnectionConfig gateway connection configuration
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder gatewayMode(GatewayConnectionConfig gatewayConnectionConfig) {
+        this.gatewayConnectionConfig = gatewayConnectionConfig;
+        return this;
+    }
+
+    /**
+     * Sets the default DIRECT connection configuration to be used.
+     *
+     * @return current CosmosClientBuilder
+     */
+    public CosmosClientBuilder directMode() {
+        this.directConnectionConfig = DirectConnectionConfig.getDefaultConfig();
```

Comment:
does this.gatewayConectionConfig get initialized somewhere else if only "directMode() is called or is it null? also for direct mode we need a different set of default values for it, not the default ones, right?

Reply:
It will be `null` - And the connection policy gets built in `buildConnectionPolicy` method - with default values for gateway configuration - with the below defaults - in `ConnectionPolicy`
```
//  Gateway connection config properties
    private int maxConnectionPoolSize = DEFAULT_MAX_POOL_SIZE;
    private Duration requestTimeout = DEFAULT_REQUEST_TIMEOUT;
    private Duration idleConnectionTimeout = DEFAULT_IDLE_CONNECTION_TIMEOUT;
```

Reply:
Yes, for `DirectConnectionConfig` - we have separate default values - which I got from `RntbdTransportClient`